### PR TITLE
Fix command spawning when spaces occur in the path

### PR DESCRIPTION
**Pull request by mvz**
## Summary

There was a bug in how `ChildProcess.build` was called. This PR fixes this.

## Details

Calling `#to_a` was called on UnixCommandString representing the executable, splitting the name unnecessarily.

## Motivation and Context

See #490.

## How Has This Been Tested?

A spec was added for this. This still needs to be tested on Windows.

## Types of changes

- Bug fix (non-breaking change which fixes an issue)

## Checklist:

- I've added tests for my code

### CODE REVIEW

**Comment by mvz** [comment]
This may need a slightly diffferent approach to not break on Windows.

**Comment by mvz** [comment]
This is ready for review.

**Comment by mvz** [comment]
@olleolleolle ok to merge?

File: lib/aruba/platforms/windows_command_string.rb [hunk 1]
```diff
@@ -11,13 +11,18 @@ module Platforms
     # @private
     class WindowsCommandString < SimpleDelegator
       def initialize(cmd)
-        __setobj__ format('%s /c "%s"', Aruba.platform.which('cmd.exe'), cmd)
+        __setobj__ cmd
```
Comment (olleolleolle) [inline lib/aruba/platforms/windows_command_string.rb]: This is the default right? No explicit need to implement the constructor?

Comment (mvz) [reply lib/aruba/platforms/windows_command_string.rb]: You may be right. I'll take a look.

Comment (mvz) [reply lib/aruba/platforms/windows_command_string.rb]: Yes, it is no longer needed. I removed it.

**Pull request merged**